### PR TITLE
feat(cat-voices): unlock password stage

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

Adds unlock password panel

## Related Issue(s)

Part of #816 

## Demo

https://github.com/user-attachments/assets/8eedccf3-0ea9-4269-90a3-b219f4cec8b5

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/pages/registration/create_keychain/stage/unlock_password_panel.dart [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+class _UnlockPasswordPanelState extends State<UnlockPasswordPanel> {
+  late final TextEditingController _passwordController;
+  late final TextEditingController _confirmPasswordController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    final registrationState = RegistrationCubit.of(context).state;
+    final createKeychain =
```
Comment (digitalheartxs) [inline catalyst_voices/lib/pages/registration/create_keychain/stage/unlock_password_panel.dart:30]: Shall we suffix it with `State` for clarity?

**Review by digitalheartxs — DISMISSED** [review]
🙌

File: catalyst_voices/lib/pages/registration/create_keychain/stage/unlock_password_panel.dart [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+class _UnlockPasswordPanelState extends State<UnlockPasswordPanel> {
+  late final TextEditingController _passwordController;
+  late final TextEditingController _confirmPasswordController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    final registrationState = RegistrationCubit.of(context).state;
```
Comment (dtscalac) [inline catalyst_voices/lib/pages/registration/create_keychain/stage/unlock_password_panel.dart:29]: Was there a valid reason to not pass this state via the constructor fields in UnlockPasswordPanel?

File: catalyst_voices/lib/pages/registration/create_keychain/stage/unlock_password_panel.dart (hunk 1 above)
Comment (damian-molinski) [reply catalyst_voices/lib/pages/registration/create_keychain/stage/unlock_password_panel.dart:30]: I'm going to refactor it soon in separate PR but yes, ill need to make it consistent.

File: catalyst_voices/lib/pages/registration/create_keychain/stage/unlock_password_panel.dart (hunk 2 above)
Comment (damian-molinski) [reply catalyst_voices/lib/pages/registration/create_keychain/stage/unlock_password_panel.dart:29]: Well, actually i just did think about it. Will do.

**Review by dtscalac — DISMISSED** [review]

**Pull request merged**